### PR TITLE
Add preliminary support for Basilisk v3

### PR DESCRIPTION
**Pull request by Theorian**

### CODE REVIEW

File: src/sample_cli.c [hunk 1]
```diff
@@ -11,7 +11,15 @@ int main(int argc, const char * argv[]) {
...
         printf("%#06x\n", device.productId);
+        if (device.productId == 0x0099){
+            printf("Found Basilisk V3\n");
+            // printf("dpi: %u\n", razer_attr_read_dpi(device.usbDevice));
 
+            razer_attr_write_scroll_mode_static(device.usbDevice, "F", 3);
+            razer_attr_write_left_mode_static(device.usbDevice, "F", 3);
+            razer_attr_write_logo_mode_static(device.usbDevice, "F", 3);
+
+        }
```
Comment (edualm) [inline src/sample_cli.c:22]: Can you revert your changes to this file? We don't really want to be adding support for specific devices on the sample cli.

File: src/lib/razermouse_driver.c [hunk 2]
```diff
@@ -1611,8 +1643,10 @@ ssize_t razer_attr_write_right_mode_reactive(IOUSBDeviceInterface **usb_dev, con
 
 ushort razer_attr_read_dpi(IOUSBDeviceInterface **usb_dev)
 {
-    struct razer_report report, response_report;
+    struct razer_report report;
+    struct razer_report response_report = {0};
```
Comment (edualm) [inline src/lib/razermouse_driver.c:1647]: Any reason for this change?

File: src/lib/razermouse_driver.c [hunk 3]
```diff
@@ -1479,9 +1517,6 @@ ssize_t razer_attr_write_logo_mode_reactive(IOUSBDeviceInterface **usb_dev, cons
 
         case USB_DEVICE_ID_RAZER_MAMBA_ELITE:
         case USB_DEVICE_ID_RAZER_BASILISK_V2:
-            report = razer_chroma_extended_matrix_effect_reactive(VARSTORE, LOGO_LED, speed, (struct razer_rgb*)&buf[1]);
```
Comment (edualm) [inline src/lib/razermouse_driver.c:1482]: Why the changes on Basilisk V2?

**Comment by edualm** [comment]
I've added some comments, can you go over them please?

File: src/lib/razermouse_driver.c (hunk 2 above)
Comment (Theorian) [reply src/lib/razermouse_driver.c:1647]: Not really just looked like it conformed to the rest of the code. but I'll revert this as well.

**Comment by Theorian** [comment]
Requested changes completed 👍

File: src/lib/razermouse_driver.c [hunk 4]
```diff
@@ -1534,9 +1572,7 @@ ssize_t razer_attr_write_scroll_mode_reactive(IOUSBDeviceInterface **usb_dev, co
 
         case USB_DEVICE_ID_RAZER_MAMBA_ELITE:
         case USB_DEVICE_ID_RAZER_BASILISK_V2:
-            report = razer_chroma_extended_matrix_effect_reactive(VARSTORE, SCROLL_WHEEL_LED, speed, (struct razer_rgb*)&buf[1]);
-            report.transaction_id.id = 0x1f;
-            break;
+
```
Comment (edualm) [inline src/lib/razermouse_driver.c:1575]: Shouldn't this be reverted too?

Comment (Theorian) [reply src/lib/razermouse_driver.c:1575]: Yes sir! Thank you for catching those.

**Review by edualm — APPROVED** [review]
